src/arrays/arraymath.jl

```diff
@@ -42,69 +46,96 @@ end
 
 # operator * operator -> operator
 function *{B<:OrthonormalBasis}(qm::QuMatrix{B}, dm::DualMatrix{B})
-    return QuArray(A_mul_Bc(rawcoeffs(qm), rawcoeffs(dm)),
-                   bases(qm,1),
-                   bases(dm,2))
+    mc = A_mul_Bc(rawcoeffs(dm), rawcoeffs(qm))
+    QAT = similar_type(typeof(qm))
```

Comment:
@acroy I guess as we know the type of `qm` which is `QuMatrix`, cant we directly return a `QuArray` ?? From my understanding it is an advantage if we use `promote_type` and `similar_type` in combination as in some of the cases below. Here as we know the type, is it possible that we pass it directly ?? As from the conversation we seem to have `QuDensity` as a subtype of `AbstractQuMatrix` so I guess we could pass `QuArray`. Please do let me know if I am missing a bigger picture.